resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/MountConfDirExecutorFeatureStep.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.k8s.features
+
+import scala.collection.JavaConverters._
+
+import io.fabric8.kubernetes.api.model.{ContainerBuilder, PodBuilder}
+
+import org.apache.spark.deploy.k8s.{KubernetesConf, SparkPod}
+import org.apache.spark.deploy.k8s.Constants.{SPARK_CONF_DIR_INTERNAL, SPARK_CONF_VOL_EXEC}
+import org.apache.spark.deploy.k8s.submit.KubernetesClientUtils
+
+/**
+ * Populate the contents of SPARK_CONF_DIR, except spark-(conf|properties) and templates to
+ * configMap to be mounted on executor pods.
+ */
+private[spark] class MountConfDirExecutorFeatureStep(conf: KubernetesConf)
```

Comment:
Just a question: Is it better to merge this into `BasicExecutorFeatureStep`?

Reply:
Hi @dongjoon-hyun , thanks for taking a look.
It can be merged with `BasicExecutorFeatureStep`, I found that step is already doing a lot of stuff.

 Do you think keeping it like this improves readability, I have no strong preference one way or the other.

Reply:
@dongjoon-hyun Ping!

Reply:
For me, I prefer to have this in `BasicExecutorFeatureStep`, @ScrapCodes . Could you update this PR into that direction please?

Reply:
Yes !